spec/text-editor-component-spec.js

```diff
@@ -894,6 +896,73 @@ describe('TextEditorComponent', () => {
       expect(component.getGutterContainerWidth()).toBe(originalGutterContainerWidth)
       expect(component.getLineNumberGutterWidth()).toBe(originalLineNumberGutterWidth)
     })
+
+    describe('randomized tests', () => {
+      let originalTimeout
+
+      beforeEach(() => {
+        originalTimeout = jasmine.getEnv().defaultTimeoutInterval
+        jasmine.getEnv().defaultTimeoutInterval = 60 * 1000
+      })
+
+      afterEach(() => {
+        jasmine.getEnv().defaultTimeoutInterval = originalTimeout
+      })
+
+      it('renders the visible rows correctly after randomly mutating the editor', async () => {
```

Comment:
Thanks so much for adding this. We probably should have had something like this all along. Lesson learned. We should never again release a system this complex without randomized testing it.